test/NuGet.Server.V2.Tests/NuGetODataControllerTests.cs

```diff
@@ -294,6 +314,646 @@ public void FindPackagesByIdDoesNotReturnDeletedPackages()
+
+                // Assert
+                Assert.Equal(4, result.Length);
+                AssertPackage(new { Id = "Foo", Version = "1.0.0" }, result[0]);
+                AssertPackage(new { Id = "Foo", Version = "1.1.0" }, result[1]);
+                AssertPackage(new { Id = "Foo", Version = "1.2.0" }, result[2]);
+                AssertPackage(new { Id = "Qux", Version = "2.0" }, result[3]);
+            }
+
+            [Fact]
+            public void GetUpdatesFiltersByTargetFramework()
+            {
+                // Arrange
+                // Arrange
+                var repo = new Mock<IServerPackageRepository>(MockBehavior.Strict);
+                repo.Setup(r => r.GetPackages()).Returns(
+                    new[]
+                {
+                        CreatePackageWithDefaults("Foo", "1.0.0", listed: true),
+                        CreatePackageWithDefaults("Foo", "1.1.0", listed: true, supportedFrameWorks: "SL5,Net40-Full"),
+                        CreatePackageWithDefaults("Foo", "1.3.0-alpha", listed: true, supportedFrameWorks: "SL5,Net40-Full"),
+                        CreatePackageWithDefaults("Foo", "2.0.0", listed: true, supportedFrameWorks: "SL5,WinRT"),
+                        CreatePackageWithDefaults("Qux", "2.0", listed: true),
+                }.AsQueryable());
+
+                var v2Service = new TestableNuGetODataController(repo.Object); v2Service.Request = new HttpRequestMessage(HttpMethod.Get, "https://localhost:8081/");
+
+                // Act
+                var result = v2Service.GetUpdates(
+                    new ODataQueryOptions<ODataPackage>(new ODataQueryContext(NuGetV2WebApiEnabler.BuildNuGetODataModel(), typeof(ODataPackage)), v2Service.Request),
+                    "Foo|Qux",
+                    "1.0|1.5",
+                    includePrerelease: false,
+                    includeAllVersions: true,
+                    targetFrameworks: "net40",
+                    versionConstraints: null)
+                    .ExpectQueryResult<ODataPackage>()
+                    .GetInnerResult()
+                    .ExpectOkNegotiatedContentResult<IQueryable<ODataPackage>>()
+                    .ToArray();
+
+                // Assert
+                Assert.Equal(2, result.Length);
+                AssertPackage(new { Id = "Foo", Version = "1.1.0" }, result[0]);
+                AssertPackage(new { Id = "Qux", Version = "2.0" }, result[1]);
+            }
+
+            [Fact]
+            public void GetUpdatesFiltersIncludesHighestPrereleasePackage()
+            {
+                // Arrange
+                var repo = new Mock<IServerPackageRepository>(MockBehavior.Strict);
+                repo.Setup(r => r.GetPackages()).Returns(
+                    new[]
+                {
+                        CreatePackageWithDefaults("Foo", "1.0.0", listed: true),
+                        CreatePackageWithDefaults("Foo", "1.1.0", listed: true, supportedFrameWorks: "SL5,Net40-Full"),
```

Comment:
The implementation in the NuGetODataController does little more than parse arguments and forward to repository.GetUpdatesCore().

This test therefore fails since GetUpdatesCore is not mocked. And mocking GetUpdatesCore kind of destroys the point of this test.